Extension/src/Utility/Process/streams.ts

```diff
@@ -0,0 +1,568 @@
+        this.streamWrite = event;
+    }
+    setWriteNotifier(notifier: (text: string) => void) {
+        this.wroteStream = notifier;
+    }
+    protected wroteStream?: (text: string) => void;
+    protected streamWrite?: (text: string) => Promise<EventStatus | string>;
+
+    constructor(stream: Duplex);
+    constructor(readable: Readable, writeable: Writable);
+    constructor(readable: Readable | Duplex, writeable?: Writable) {
+        super(readable);
+        this.writeable = writeable || readable as Writable;
+
+        this.writeable.on('error', (_error) => {
+            /*
+            if ((global as any).DEVMODE && error) {
+              verbose(`write-stream - error - ${error.message}`);
+            }
+            */
+        });
+    }
+
+    async write(...text: string[]): Promise<void> {
+        let content = text;
+
+        if (this.streamWrite) {
+            content = new Array<string>();
+
+            // allow the writing to be intercepted
+            for (const each of text) {
+                if (each) {
+                    const result = await this.streamWrite(each);
+                    if (!is.cancelled(result)) {
+                        content.push(result || each);
+                    }
+                }
+            }
+        }
+        if (this.wroteStream) {
+            for (const each of content) {
+                this.wroteStream(each);
+            }
+        }
+
+        return new Promise((resolve, reject) => {
+            // send the content to the stream
+            this.writeable.write(this.#encoder.encode(content.join('')), (error: Error | null | undefined) => {
+                if (error) {
+                    reject(error);
+                } else {
+                    resolve();
+                }
+            });
+        });
+    }
+
+    async writeln(...text: string[]): Promise<void> {
+        let content = text;
+
+        if (this.streamWrite) {
+            content = new Array<string>();
+
+            // allow the writing to be intercepted
+            for (const each of text) {
+                if (each) {
+                    const result = await this.streamWrite(each);
+                    if (!is.cancelled(result)) {
+                        content.push(result || each);
+                    }
+                }
+            }
+        }
+
+        if (this.wroteStream) {
+            for (const each of content) {
+                this.wroteStream(each);
+            }
+        }
+
+        return new Promise((resolve) => {
+            // send the content to the stream
+            try {
+                if (!this.writeable.destroyed) {
+                    this.writeable.write(this.#encoder.encode(content.join('\n') + '\n'), (error: Error | null | undefined) => {
+                        if ((global as any).DEVMODE && error) {
+                            verbose(`stream-closed - ${error.message}`);
+                            // reject(error);

```

Comment:
Is this `// reject(error);` supposed to be here? Same as the case below on 527. If so, could you add a comment?

Reply:
removed.